doc/source/developers_guide.rst

```diff
@@ -294,10 +296,13 @@ there is no name associated with such a program, the corresponding
 symbol table is given the name "fparser2:main_program", chosen so as
 to prevent any clashes with other Fortran names.
```

Comment:
Is part of this paragraph out of date now?
It says: "This is handled for all cases bar one within the `BlockBase`" but there are several classes inheriting BlockBase that are not ScopingRegionMixin e.g. IfConstruct, WhereConstruct, ForallConstrust, ...(they also werent before because of the explicit test.)
Can this be simplified?

Reply:
As discussed, I think the text is correct. Although not all subclasses of `BlockBase` represent scoping regions, all scoping regions *are* subclasses of BlockBase so that's where we handle the enter/exit task.